sympy/ntheory/elliptic_curve.py

```diff
@@ -76,7 +76,9 @@ def __contains__(self, point):
             raise ValueError('Invalid point.')
         if self.characteristic == 0 and z1 == 0:
             return True
-        return self._eq.subs({self.x: x1, self.y: y1, self.z: z1})
+        values = {self.x: x1, self.y: y1, self.z: z1}
+        val = self._eq.lhs.subs(values) - self._eq.rhs.subs(values)
```

Comment:
I think that lhs and rhs can just be made into baked lambda functions, rather than relying on subs.

Reply:
`self._eq` could probably be a `Poly` object. Then its value is obtained by `self._eq(x1, y1, z1)` and the result is an object of `self._domain`.